apps/meteor/server/models/raw/Sessions.ts

```diff
@@ -178,115 +178,103 @@ export const aggregates = {
+
+					// we filter by createdAt to be able to perform a range query, since the "day", "month" and "year" fields were not meant for that
+					createdAt: { $gte: from, $lt: to },
 				},
-				{
-					$group: {
-						_id: {
-							userId: '$userId',
-							device: '$device',
-							day: '$day',
-							month: '$month',
-							year: '$year',
-						},
-						mostImportantRole: { $first: '$mostImportantRole' },
-						time: { $sum: '$time' },
-						sessions: { $sum: 1 },
-					},
+			},
+			{
+				$project: {
+					userId: 1,
+					device: 1,
+					day: 1,
+					month: 1,
+					year: 1,
+					mostImportantRole: 1,
+					time: { $trunc: { $divide: [{ $subtract: ['$lastActivityAt', '$loginAt'] }, 1000] } },
 				},
-				{
-					$sort: {
-						time: -1,
-					},
+			},
+			{
+				$match: {
+					time: { $gt: 0 },
 				},
-				{
-					$group: {
-						_id: {
-							userId: '$_id.userId',
-							day: '$_id.day',
-							month: '$_id.month',
-							year: '$_id.year',
-						},
-						mostImportantRole: { $first: '$mostImportantRole' },
-						time: { $sum: '$time' },
-						sessions: { $sum: '$sessions' },
-						devices: {
-							$push: {
-								sessions: '$sessions',
-								time: '$time',
-								device: '$_id.device',
-							},
-						},
+			},
+			{
+				$group: {
+					_id: {
+						userId: '$userId',
+						device: '$device',
+						day: '$day',
+						month: '$month',
+						year: '$year',
 					},
+					mostImportantRole: { $first: '$mostImportantRole' },
+					time: { $sum: '$time' },
+					sessions: { $sum: 1 },
 				},
-				{
-					$sort: {
-						_id: 1,
-					},
+			},
+			{
+				$sort: {
+					time: -1,
 				},
-				{
-					$project: {
-						_id: 0,
-						type: { $literal: 'user_daily' },
-						_computedAt: { $literal: new Date() },
+			},
+			{
+				$group: {
+					_id: {
+						userId: '$_id.userId',
 						day: '$_id.day',
 						month: '$_id.month',
 						year: '$_id.year',
-						userId: '$_id.userId',
-						mostImportantRole: 1,
-						time: 1,
-						sessions: 1,
-						devices: 1,
+					},
+					mostImportantRole: { $first: '$mostImportantRole' },
+					time: { $sum: '$time' },
+					sessions: { $sum: '$sessions' },
+					devices: {
+						$push: {
+							sessions: '$sessions',
+							time: '$time',
+							device: '$_id.device',
+						},
 					},
 				},
-			],
-			{ allowDiskUse: true },
-		);
+			},
+			{
+				$sort: {
+					_id: 1,
+				},
+			},
+			{
+				$project: {
+					_id: 0,
+					type: { $literal: 'user_daily' },
+					_computedAt: { $literal: new Date() },
+					day: '$_id.day',
+					month: '$_id.month',
+					year: '$_id.year',
+					userId: '$_id.userId',
+					mostImportantRole: 1,
+					time: 1,
+					sessions: 1,
+					devices: 1,
+				},
+			},
+		];
+
+		return collection.aggregate<
+			Pick<ISession, 'mostImportantRole' | 'userId' | 'day' | 'year' | 'month' | 'type'> & {
+				time: number;
+				sessions: number;
+				devices: ISession['device'][];
+				_computedAt: string;
+			}
```

Comment:
This should be its own type, since you're reusing it above

Reply:
👍 we can do that later